server/jms/src/main/java/org/hawkular/apm/server/jms/RetryCapableMDB.java

```diff
@@ -110,6 +147,30 @@ public void onMessage(Message message) {
      * @param retryCount The remaining retry count
      * @throws Failed to process items
      */
-    protected abstract void process(String tenantId, List<S> items, int retryCount) throws Exception;
+    protected void process(String tenantId, List<S> items, int retryCount) throws Exception {
+        ProcessingUnit<S, T> pu =
+                new ProcessingUnit<S, T>();
+
+        pu.setProcessor(getProcessor());
+        pu.setRetryCount(retryCount);
+
+        pu.setResultHandler(new Handler<T>() {
```

Comment:
What's the minimum Java version for this module? I think it would be more readable by using a lambda here.